feat(etc) : Configuring the package local test environment

XionWCFM opened the pull request:
close #68 

# Features

Now we can test our packages directly in our local environment before deploying them.

This PR implements the suggestion in #68 

# Usage

You can access tests configured with playwright via apps/app-router and apps/pages-router pnpm run e2e.

```
pnpm run build
// cd apps/app-router or cd apps/pages-router
pnpm run e2e
```

Because it depends on the build results of the local environment, you must build the packages before running them.

I think it would be more convenient to control all command usage at the root level, but I did not do it because it is beyond the scope of this PR.

# Review

in packages/core/package.json
in packages/next/package.json

```json
  "main": "./src/index.ts", // before
  "main": "./dist/index.js", // after
  "publishConfig": {
```

If the entry point was not set to look at the build output, the package could not be imported properly in the monorepo environment, so this had to be modified.

# etc

I forgot the conventional commit rules of use-funnel and ended up committing with gitmoji.

I think this can be solved by doing a squash merge. If you merge, please do a squash merge. 🥹

minuukang commented on pnpm-workspace.yaml:
```diff
@@ -1,3 +1,4 @@
 packages:
   - packages/*
   - docs
+  - apps/*
```
The "apps" directory seems to create a service that derives from something `@use-funnel`, what about a name like "example"?

XionWCFM replied:
>I agree, I made it idiomatic without much thought.


I think it would be better to use tests rather than examples. 

Code and examples for e2e testing may initially be similar, but

As testing becomes more sophisticated, it may develop into examples that are too complicated for users.

What do you think?

minuukang replied:
I agree. One of my plans was to include examples of different router adapters, and it would be great to have e2e tests using them. But as you say, if it gets complicated, it would be nice to separate it with a name like "e2e_test"!

minuukang commented on apps/app-router/package.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "app-router",
```
```suggestion
  "name": "nextjs-app-router",
```

We can provide examples with other adapters, not just Next.js, so it's probably a good idea to add a prefix to the package and folder names.

XionWCFM replied:
I missed that part 🔥, thank you! I think it would be a good idea to change the folder name as well.

I'll work on it

minuukang commented on apps/app-router/e2e/app-router-funnel.spec.ts:
```diff
@@ -0,0 +1,15 @@
+
+  await page.getByRole('button', { name: 'next' }).click();
+
+  await expect(page.getByText('end')).toBeVisible();
```
It would be nice to have a test case for browser "history back".

XionWCFM replied:
🔥 That's a good idea! I will also add a test case for history.back.

minuukang commented:
I'll have to prepare a CI to run unit tests and e2e tests on this repo!

XionWCFM commented:
> I'll have to prepare a CI to run unit tests and e2e tests on this repo!

I think it would be okay to add CI to this PR as well.

This is an additional suggestion, but I think it would be good to add the process of building an application that uses a funnel in the CI process.

That way you can find cases of build failures like the current @use-funnel/browser before deploying the package.

However, if you add this test now, CI will be broken due to the use case of the app router, so it would be better to add it later.

XionWCFM commented on examples/nextjs-app-router/package.json:
```diff
@@ -0,0 +1,28 @@
+{
+	"name": "nextjs-app-router",
```
Nextjs prefix was added and the folder name was modified to match the name.

minuukang reviewed (commented):
Great! Finally, I think I only need to remove package-lock.json and regenerate pnpm-lock!

XionWCFM commented:
> 좋아요! 마지막으로 package-lock.json만 제거하고 pnpm-lock을 재생성하면 될 것 같아요!

It was a problem I wasn't aware of. Thank you!
Removed package.lock.json and updated pnpm.lock.
